persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickProvider.scala

```diff
@@ -38,12 +41,33 @@ private[lagom] class SlickProvider(
 
   val autoCreateTables: Boolean = createTables.getBoolean("auto")
 
+  private val asyncExecConfig =
+    new AsyncExecutorConfig(system.settings.config.getConfig("jdbc-defaults.slick.async-executor"))
+
   if (dbApi != null) {
     // Work around https://github.com/playframework/playframework/issues/7262
     // Set the system property
     System.setProperty(Context.PROVIDER_URL, "/")
-    // Rebind the datasource
-    new InitialContext().rebind("DefaultDS", dbApi.database("default").dataSource)
+
+    // the data source as configured by Play
+    val dataSource = dbApi.database("default").dataSource
+
+    // Slick Database configured with Play DS + AsyncExecutor
+    val database =
+      Database.forDataSource(
+        ds = dbApi.database("default").dataSource,
+        maxConnections = Option(asyncExecConfig.maxConnections),
+        executor = AsyncExecutor(
+          name = "AsyncExecutor.default",
+          minThreads = asyncExecConfig.minConnections,
+          maxThreads = asyncExecConfig.numThreads,
+          queueSize = asyncExecConfig.queueSize,
+          maxConnections = asyncExecConfig.maxConnections
+        )
+      )
+
+    // bind the DB configured with an AsyncExecutor
+    new InitialContext().rebind("DefaultDB", database)
```

Comment:
We don't want to bind the Play DS to JNDI, otherwise akka-persistence-plugin will pick it instead. We want the `jndiName` lookup to return `None` and force all DB usages to pick the one we are adding to JNDI.

Reply:
BTW, I believe this is safe. The persistence component is the only one, as of today, that depends on `ClusterComponent`.  

The akka plugin only gets used for the first time when the cluster starts up. At that point, the Database is already available as a JNDI resource.